core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts

```diff
@@ -196,4 +198,30 @@ export class DotPagesComponent implements AfterViewInit, OnDestroy {
             menuDOMID.includes('pageActionButton') || menuDOMID.includes('favoritePageActionButton')
         );
     }
+
+    /**
+     * Subscribe to the shutdown event of the contentlet dialog
+     *
+     * @param {*} componentRef
+     * @return {*}
+     * @memberof DotPagesComponent
+     */
+    subscribeToShutdown(componentRef: Component): void {
+        if (!(componentRef instanceof DotCreateContentletComponent)) return;
+
+        this.contentletDialogShutdown = componentRef.shutdown.subscribe(() => {
+            this.store.getPages({
+                offset: 0
+            });
+        });
+    }
+
+    /**
+     * Unsubscribe to the shutdown event of the contentlet dialog
+     *
+     * @memberof DotPagesComponent
+     */
+    unsubscribeToShutdown() {
+        if (this.contentletDialogShutdown) this.contentletDialogShutdown.unsubscribe();
+    }
```

Comment:
Found difficult to test this, since router and store are mocked in the tests, if you find a way to test this in a non destructive way, we can try to implement it

Reply:
```
    fit('test router outlet', () => {
        const routerOutlet = de.query(By.css('router-outlet'));
        console.log(routerOutlet);
        routerOutlet.triggerEventHandler('activate', {});
        expect(routerOutlet).toBeTruthy();
    })
```

This will trigger events but not sure how to pass the instance of `DotCreateContentletComponent` but pretty sure we can.